feat(Microsoft SharePoint Node): Allow filtering on non-indexed columns

pemontto opened the pull request:
## Summary

Adds the `Prefer: HonorNonIndexedQueriesWarningMayFailRandomly` header to `item:getAll` operations when a filter is applied.

This allows users to filter on list columns that are not indexed in SharePoint (like `FileDirRef` and `FileLeafRef`), which would otherwise result in an error from the Microsoft Graph API.

![Pasted image 20250709143030-opt](https://github.com/user-attachments/assets/5b95cb63-e5c1-4770-9de5-d54f67bc34c2)

## Related Linear tickets, Github issues, and Community forum posts

- None

## Review / Merge checklist

- [X] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md))
- [X] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

pemontto committed:
feat(SharePoint Node): allow filtering on non-indexed columns

Adds the `Prefer: HonorNonIndexedQueriesWarningMayFailRandomly` header to `item:getAll` operations when a filter is applied.

This allows users to filter on list columns that are not indexed in SharePoint, which would otherwise result in an error from the Microsoft Graph API.

Joffcom commented:
Hey @pemontto,

Thanks for the PR, We have created "GHC-3017" as the internal reference to get this reviewed.

One of us will be in touch if there are any changes needed, in most cases this is normally within a couple of weeks but it depends on the current workload of the team.

pemontto committed:
fix(SharePoint Node): avoid overwriting existing Prefer header

The previous implementation would replace any existing `Prefer` header when allowing non-indexed queries.

This change ensures that the `HonorNonIndexedQueriesWarningMayFailRandomly` value is appended, preserving other preferences that might have been set.
